crates/core/types/filters.rs

```diff
@@ -0,0 +1,99 @@
+use ethereum_rust_rlp::{
+    decode::{decode_rlp_item, RLPDecode},
+    encode::RLPEncode,
+    error::RLPDecodeError,
+    structs::{Decoder, Encoder},
+};
+use ethereum_types::{Address, H160, H256};
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(untagged)]
+/// Address filter used to filter Logs.
+pub enum AddressFilter {
+    Single(H160),
+    Many(Vec<H160>),
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(untagged)]
+/// Topic filter used to filter Logs.
+pub enum TopicFilter {
+    Topic(H256),
+    Topics(Vec<H256>),
+}
+
+#[derive(Debug, Clone)]
+pub struct LogsFilter {
```

Comment:
is this structure equivalent to the one you're using for getLogs?

Reply:
It's similar, I had to move it here to avoiding having circular dependencies, also I removed the options to declutter the code which will use this structure.

Reply:
I think I'll remove it, since now we'll be using a hashmap instead of using the db, and I won't need it for the storage crate.

Reply:
Removed!